doc/external-libs/ginkgo.html

```diff
@@ -0,0 +1,98 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
+	  "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+  <head>
+    <title>The deal.II Readme on interfacing to Ginkgo</title>
+    <link href="../screen.css" rel="StyleSheet">
+    <meta name="copyright" content="Copyright (C) 2008 - 2018 by the deal.II authors">
+    <meta name="keywords" content="deal.II">
+  </head>
+
+  <body>
+
+    <h1>Interfacing <acronym>deal.II</acronym> to Ginkgo</h1>
+
+    <p>
+        <a href="https://github.com/ginkgo-project/ginkgo"
+           target="_top">Ginkgo</a> is a
+        Numerical linear algebra software package that provides its users with
+        highly optimized fine grid level linear algebra operations. It currently has
+        the capability to solve on the CPU with the support of OpenMP and on the GPU
+        with NVIDIA's cuda libraries. For some operations it uses NVIDIA's own libraries, CuSparse
+        and CuBLAS but some routines also have self implemented cuda functions. See the documentation
+        for more details.
+    </p>
+
+    <p>
+      <acronym>deal.II</acronym> has wrapper classes to the linear algebra
+      parts of Ginkgo that provide almost the
+      same interfaces as the built-in <acronym>deal.II</acronym> linear algebra operations.
+    </p>
+
+    <h2>Installing <acronym>deal.II</acronym> with Ginkgo</h2>
+
+    <p>
+        During the cmake configuration, the following flags should be specified
+        <code>-DDEAL_II_WITH_GINKGO=ON</code>, to enable Ginkgo and the <code>-DGINKGO_DIR=DIR</code>
+        can be set with the path where Ginkgo has been installed. 
+    </p>
+
+    <h3>Installing Ginkgo</h3>
+
+    <p>
+        Installing Ginkgo is quite simple. Ginkgo currently uses Cmake and hence one can use
+        the following commands to easily install Ginkgo. For different dependencies and compatible
+        libraries, please refer to <a href="https://github.com/ginkgo-project/ginkgo"
+                                      target="_top">Ginkgo</a>.
+        The different flags that can be added are:
+        <code> -DBUILD_REFERENCE={ON,OFF}:</code> Builds the reference single thread implementation
+        used to check the more sophisticated implementations of the GPU/CPU.
+        <code> -DBUILD_CUDA={ON,OFF}:</code> Builds the GPU implementation specifically the NVIDIA
+        cuda implementations. This needs CUDA to be installed on the machine.
+        <code> -DBUILD_OMP={ON,OFF}:</code> Builds the multi-threaded OpenMP implementations.
```

Comment:
this is relevant to https://github.com/dealii/dealii/issues/7300

I am not sure one should recommend disabling `openmp`. I had bad experience when `openmp` was enabled in `openblas` together with `tbb` active in `deal.II` (and I was NOT calling `blas` functions from TBB-parallel part of `dealii`/user code). But this could be specific to `openblas` implementation of multithreading...

Could you maybe tests that you don't see any slow-downs when `omp` is enabled in `gingo`?

Reply:
Thank you for your comments. Hm, this is interesting. Do you suggest any specific timed examples or tests that you have which I can use ?

Reply:
Thanks. I have tested this. I do not observe any slowdown and I see an appropriate scaling for large enough problem sizes. I do not expect this to be a problem in the future (as long as the interface to ginkgo remains the same) as well because during the solution phase when deal.ii gives the control over to ginkgo, there shouldn't be any threads "working" within deal.ii which might interfere with the Omp threads spawned by Ginkgo.

Reply:
Thanks for checking this